soc/intel/common/soc_gpio.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/acpi/acpi.h>
+#include "soc.h"
+#include "soc_gpio.h"
+
+static int gpio_info_acpi_get(uint32_t bank_idx, uint32_t field_idx, uint32_t *ret_val)
+{
+	int status;
+	ACPI_OBJECT args[2];
+	ACPI_OBJECT_LIST arg_list;
+	ACPI_OBJECT gpio_obj;
+
+	arg_list.Count = 2;
+	arg_list.Pointer = args;
+
+	args[0].Type = ACPI_TYPE_INTEGER;
+	args[0].Integer.Value = bank_idx;
+
+	args[1].Type = ACPI_TYPE_INTEGER;
+	args[1].Integer.Value = field_idx;
+
+	status = acpi_invoke_method("\\_SB.GINF", &arg_list, &gpio_obj);
```

Comment:
Should the `\\_SB.GINF` perhaps be a define? Would it make sense to group standard identifiers into some common header file?